### PR TITLE
Bluetooth: Mesh: Add option to call update cb on every retransmission

### PR DESCRIPTION
**Pull request by PavelVPV**
A user may want to control message parameters (for example, delay
parameter) on every retransmission of a published message (for example,
see section 1.4.1 of the mesh model specification). This is essential
for lighting messages as time gap between messages retransmitted via
the publish-retransmit mechanism introduces unwanted jitter/pop-corn
when such retransmissions are received by a large 'group' of lights.

This commit adds an option to `struct bt_mesh_model_pub` to make the
access layer call `bt_mesh_model_pub.update` callback on every
retransmission. This also addes few macros and functions that can be
used for further calculations.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

### CODE REVIEW

**Review by Andrewpini — APPROVED** [review]
LGTM 👍

File: subsys/bluetooth/mesh/access.c
Comment (omkar3141) [inline subsys/bluetooth/mesh/access.c:247]: This looks fine. However, it is not obvious how this `update` callback can be used to introduce changing parameters in retransmissions.  Description of the `update` needs to be updated to tell how retransmission can be detected against the regular periodic publication (and its retransmission itself). Basically by paraphrasing what is shown here.

**Review by omkar3141 — CHANGES_REQUESTED** [review]

**Review by omkar3141 — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Comment by PavelVPV** [comment]
Hi, @jhedberg , would you be able look at this PR? I need one more approval from a code owner to get this merged.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
